Description: Fix prototypes for xattr functions to match sys/xattr.h
Author: Adam Conrad <adconrad@ubuntu.com>

--- libfakeroot.c
+++ libfakeroot.c
@@ -1661,7 +1661,7 @@
 #endif /* HAVE_CAPSET */
 
 #if defined(HAVE_SETXATTR) || defined(HAVE_LSETXATTR) || defined(HAVE_FSETXATTR)
-static size_t common_setxattr(INT_STRUCT_STAT *st, const char *name, void * value, size_t size, int flags)
+static int common_setxattr(INT_STRUCT_STAT *st, const char *name, const void *value, size_t size, int flags)
 {
   xattr_args xattr;
   xattr.name = name;
@@ -1716,7 +1716,7 @@
 #endif /* defined(HAVE_LISTXATTR) || defined(HAVE_LLISTXATTR) || defined(HAVE_FLISTXATTR) */
 
 #if defined(HAVE_REMOVEXATTR) || defined(HAVE_LREMOVEXATTR) || defined(HAVE_FREMOVEXATTR)
-static size_t common_removexattr(INT_STRUCT_STAT *st, const char *name)
+static int common_removexattr(INT_STRUCT_STAT *st, const char *name)
 {
   xattr_args xattr;
   xattr.name = name;
@@ -1734,7 +1734,7 @@
 #endif /* defined(HAVE_REMOVEXATTR) || defined(HAVE_LREMOVEXATTR) || defined(HAVE_FREMOVEXATTR) */
 
 #ifdef HAVE_SETXATTR
-ssize_t setxattr(const char *path, const char *name, void *value, size_t size, int flags)
+int setxattr(const char *path, const char *name, const void *value, size_t size, int flags)
 {
   INT_STRUCT_STAT st;
   int r;
@@ -1755,7 +1755,7 @@
 #endif /* HAVE_SETXATTR */
 
 #ifdef HAVE_LSETXATTR
-ssize_t lsetxattr(const char *path, const char *name, void *value, size_t size, int flags)
+int lsetxattr(const char *path, const char *name, const void *value, size_t size, int flags)
 {
   INT_STRUCT_STAT st;
   int r;
@@ -1776,7 +1776,7 @@
 #endif /* HAVE_LSETXATTR */
 
 #ifdef HAVE_FSETXATTR
-ssize_t fsetxattr(int fd, const char *name, void *value, size_t size, int flags)
+int fsetxattr(int fd, const char *name, const void *value, size_t size, int flags)
 {
   INT_STRUCT_STAT st;
   int r;
@@ -1923,7 +1923,7 @@
 #endif /* HAVE_FLISTXATTR */
 
 #ifdef HAVE_REMOVEXATTR
-ssize_t removexattr(const char *path, const char *name)
+int removexattr(const char *path, const char *name)
 {
   INT_STRUCT_STAT st;
   int r;
@@ -1944,7 +1944,7 @@
 #endif /* HAVE_REMOVEXATTR */
 
 #ifdef HAVE_LREMOVEXATTR
-ssize_t lremovexattr(const char *path, const char *name)
+int lremovexattr(const char *path, const char *name)
 {
   INT_STRUCT_STAT st;
   int r;
@@ -1965,7 +1965,7 @@
 #endif /* HAVE_LREMOVEXATTR */
 
 #ifdef HAVE_FREMOVEXATTR
-ssize_t fremovexattr(int fd, const char *name)
+int fremovexattr(int fd, const char *name)
 {
   INT_STRUCT_STAT st;
   int r;
--- wrapfunc.inp
+++ wrapfunc.inp
@@ -169,22 +169,22 @@
 fgetxattr;ssize_t;(int fd, const char *name, void *value, size_t size);(fd, name, value, size)
 #endif /* HAVE_FGETXATTR */
 #ifdef HAVE_SETXATTR
-setxattr;ssize_t;(const char *path, const char *name, void *value, size_t size, int flags);(path, name, value, size, flags)
+setxattr;int;(const char *path, const char *name, const void *value, size_t size, int flags);(path, name, value, size, flags)
 #endif /* HAVE_SETXATTR */
 #ifdef HAVE_LSETXATTR
-lsetxattr;ssize_t;(const char *path, const char *name, void *value, size_t size, int flags);(path, name, value, size, flags)
+lsetxattr;int;(const char *path, const char *name, const void *value, size_t size, int flags);(path, name, value, size, flags)
 #endif /* HAVE_LSETXATTR */
 #ifdef HAVE_FSETXATTR
-fsetxattr;ssize_t;(int fd, const char *name, void *value, size_t size, int flags);(fd, name, value, size, flags)
+fsetxattr;int;(int fd, const char *name, const void *value, size_t size, int flags);(fd, name, value, size, flags)
 #endif /* HAVE_FSETXATTR */
 #ifdef HAVE_REMOVEXATTR
-removexattr;ssize_t;(const char *path, const char *name);(path, name)
+removexattr;int;(const char *path, const char *name);(path, name)
 #endif /* HAVE_REMOVEXATTR */
 #ifdef HAVE_LREMOVEXATTR
-lremovexattr;ssize_t;(const char *path, const char *name);(path, name)
+lremovexattr;int;(const char *path, const char *name);(path, name)
 #endif /* HAVE_LREMOVEXATTR */
 #ifdef HAVE_FREMOVEXATTR
-fremovexattr;ssize_t;(int fd, const char *name);(fd, name)
+fremovexattr;int;(int fd, const char *name);(fd, name)
 #endif /* HAVE_FREMOVEXATTR */
 
 #ifdef HAVE_FSTATAT
